packages/client/hmi-client/src/services/goLLM.ts

```diff
@@ -0,0 +1,63 @@
+import API, { SSEHandler, SSEStatus } from '@/api/api';
+import type { TaskResponse } from '@/types/Types';
+import { TaskStatus } from '@/types/Types';
+import { logger } from '@/utils/logger';
+
+/**
+ * Fetches model card data from the server.
+ * @param {string} documentId - The document ID.
+ * @param {string} modelId - The model ID.
+ * @returns {Promise<TaskResponse|null>} A promise resolving to the task response or null if unsuccessful.
+ */
+export async function modelCard(documentId: string, modelId: string): Promise<TaskResponse | null> {
+	try {
+		const response = await API.post('/gollm/model_card', null, {
+			params: {
+				'document-id': documentId,
+				'model-id': modelId
+			}
+		});
+
+		// FIXME: I think we need to refactor the response interceptors so that we can handle errors here, or even in the interceptor itself...might be worth a discussion
+		if (response.status !== 200) {
+			switch (response.status) {
+				case 404: {
+					logger.error('The provided model or document arguments are not found');
+					break;
+				}
+				case 500:
+				default: {
+					logger.error('There was an issue dispatching the request');
+					break;
+				}
+			}
+			return null;
+		}
```

Comment:
You are right those should be part of the API interceptors, feel free to remove it. we can table that discussion another day.